app/com/linkedin/drelephant/mapreduce/heuristics/GenericMemoryHeuristic.java

```diff
@@ -104,23 +115,31 @@ public HeuristicResult apply(MapReduceApplicationData data) {
     }
 
     String containerSizeStr = data.getConf().getProperty(_containerMemConf);
-    if (containerSizeStr == null) {
-      return null;
-    }
-
-    long containerMem;
-    try {
-      containerMem = Long.parseLong(containerSizeStr);
-    } catch (NumberFormatException e) {
-      // Some job has a string var like "${VAR}" for this config.
-      if(containerSizeStr.startsWith("$")) {
-        String realContainerConf = containerSizeStr.substring(containerSizeStr.indexOf("{")+1,
-            containerSizeStr.indexOf("}"));
-        containerMem = Long.parseLong(data.getConf().getProperty(realContainerConf));
-      } else {
-        throw e;
+    long containerMem = -1L;
+
+    if (containerSizeStr != null) {
+      try {
+        containerMem = Long.parseLong(containerSizeStr);
+      } catch (NumberFormatException e0) {
+        // Some job has a string var like "${VAR}" for this config.
+        if(containerSizeStr.startsWith("$")) {
+          String realContainerConf = containerSizeStr.substring(containerSizeStr.indexOf("{")+1,
+              containerSizeStr.indexOf("}"));
+          String realContainerSizeStr = data.getConf().getProperty(realContainerConf);
+          try {
+            containerMem = Long.parseLong(data.getConf().getProperty(realContainerConf));
+          }
+          catch (NumberFormatException e1) {
+            logger.warn(realContainerConf + ": expected number [" + realContainerSizeStr + "]");
+          }
+        } else {
+          logger.warn(_containerMemConf + ": expected number [" + containerSizeStr + "]");
```

Comment:
I have only a slight preference to not include the exception.  We already know it's a NumberFormatExpection (in the code) and the string that failed parsing is in the message.  I suspect that if this exception is reached, it'll likely fail repeatedly (with every job) - printing a stacktrace with every error might make it harder to debug unrelated issues.